creator-node/src/snapbackSM.js

```diff
@@ -449,71 +449,67 @@ class SnapbackSM {
   }
 
   /**
-   * Track an ongoing sync operation for a given secondaryUrl and user wallet
-   *  - Will re-enqueue sync if it fails or is still behind after retries or max duration
+   * Monitor an ongoing sync operation for a given secondaryUrl and user wallet
+   * Return boolean indicating if an additional sync is required
+   *
+   * Polls secondary for MaxSyncMonitoringDurationInMs
    */
-  async monitorSecondarySync (userWallet, primaryClockValue, secondaryUrl, syncType) {
-    const startTime = Date.now()
+  async additionalSyncIsRequired (userWallet, primaryClockValue, secondaryUrl, syncType) {
+    const MaxExportClockValueRange = this.nodeConfig.get('maxExportClockValueRange')
+    const logMsgString = `additionalSyncIsRequired (${syncType}): wallet ${userWallet} secondary ${secondaryUrl} primaryClock ${primaryClockValue}`
 
-    // Define axios request object for secondary sync status request
-    const syncMonitoringRequestParameters = {
+    // Define axios request object for secondary clock status request
+    const clockStatusRequestParams = {
       method: 'get',
       baseURL: secondaryUrl,
-      url: `/sync_status/${userWallet}`,
+      url: `/users/clock_status/${userWallet}`,
       responseType: 'json'
     }
 
-    // TODO syncAttemptCompleted does nothing - refactor this
-    let syncAttemptCompleted = false
-    let secondaryClockValAfterSync = null
-    while (!syncAttemptCompleted) {
+    const startTimeMs = Date.now()
+    const endTimeMs = startTimeMs + MaxSyncMonitoringDurationInMs
+
+    let additionalSyncRequired = true
+    while (Date.now() < endTimeMs) {
       try {
-        const syncMonitoringResp = await axios(syncMonitoringRequestParameters)
+        const clockStatusResp = await axios(clockStatusRequestParams)
+        const { clockValue: secondaryClockValue } = clockStatusResp.data.data
 
-        const respData = syncMonitoringResp.data.data
-        this.log(`processSync ${userWallet} secondary response: ${JSON.stringify(respData)}`)
+        this.log(`${logMsgString} secondaryClock ${secondaryClockValue}`)
 
-        // A success response does not necessarily guarantee completion, validate response data to confirm
-        // Returned secondaryClockValue can be greater than the cached primaryClockValue if a client write was initiated
-        //    after primaryClockValue cached and resulting sync is monitored
-        if (respData.clockValue >= primaryClockValue) {
-          secondaryClockValAfterSync = respData.clockValue
+        /**
+         * One sync op can process at most MaxExportClockValueRange range
+         * A larger clock diff will require multiple syncs; short-circuit monitoring
+         */
+        if (secondaryClockValue + MaxExportClockValueRange < primaryClockValue) {
```

Comment:
is it possible to get here too early? like we cannot guarantee that the sync has started on the other side. could it be possible that the sync gets queued or something and clock status returns the existing clock number?

Reply:
one way to fix this is plumb the secondary values to this function and check that it's greater than the current secondary value. https://github.com/AudiusProject/audius-protocol/blob/6b38f217ff88410bfb0a05802878cfc6f035d4ce/creator-node/src/snapbackSM.js#L405-L406

one thing we also may need to change with this is the dedupe needs to be able to override the max block number for the sync job

Reply:
after discussion - this is a no-op. the above condition handles only the case where a secondary could not catch up after a sync due to MaxExportClockValueRange. the default case of secondary sync monitoring does not enter this case

resolving